agent/tool-scripts/postprocess/pidstat-postprocess

```diff
@@ -149,35 +162,36 @@ while ($line = <PIDSTAT_TXT>) {
 		$pidstat{memory_usage}{virtual_size}{$pid}{$timestamp_ms} = $vsz;
 		$pidstat{memory_usage}{resident_set_size}{$pid}{$timestamp_ms} = $rss;
 		$timestamps{$timestamp_ms}++;
-	}
-}
-close(PIDSTAT_TXT);
 
-# fill in any missing data with zeros
-my $htmlpage;
-my $graph;
-my $pid;
-my $timestamp_ms;
-foreach  $htmlpage ( keys %pidstat ) {
-	foreach  $graph ( keys %{ $pidstat{$htmlpage} } ) {
-		foreach $pid ( keys %{ $pidstat{$htmlpage}{$graph} } ) {
-			foreach $timestamp_ms ( sort {$a <=> $b} (keys %timestamps ) ) {
-				if (! defined($pidstat{$htmlpage}{$graph}{$pid}{$timestamp_ms})) {
-					$pidstat{$htmlpage}{$graph}{$pid}{$timestamp_ms} = 0;
-				}
-			}
-		}
+    # parse data every 10,000 lines.
+    if ($lines_scanned % $lines_limit == 0) {
+
+        # TODO: this code shold be multithreaded in order to speed up the process.
+
+        # fill in any missing data with zeros
+        my $htmlpage;
+        my $graph;
+        my $pid;
+        my $timestamp_ms;
+        foreach  $htmlpage ( keys %pidstat ) {
+        	foreach  $graph ( keys %{ $pidstat{$htmlpage} } ) {
+        		foreach $pid ( keys %{ $pidstat{$htmlpage}{$graph} } ) {
+        			foreach $timestamp_ms ( sort {$a <=> $b} (keys %timestamps ) ) {
+        				if (! defined($pidstat{$htmlpage}{$graph}{$pid}{$timestamp_ms})) {
+        					$pidstat{$htmlpage}{$graph}{$pid}{$timestamp_ms} = 0;
+        				}
+        			}
+        		}
+        	}
+        }
+
+        # dump data into csv.
+        gen_data(\%pidstat, \%graph_type, \%graph_threshold, $dir);
```

Comment:
Won't we need to modify `gen_data()` to add the data to existing `.csv`, `.html`, and `.json` files?

Reply:
I also ask my self that question I guess @atheurer can answer that better.
AFAI look at the code looks like it's just append, since the file name was always be the same.
moreover I've did some simple test, and compare this patch with the original one.
and the csv output was with the same amount of lines.

Reply:
It seems this line, https://github.com/distributed-system-analysis/pbench/blob/master/agent/tool-scripts/postprocess/GenData.pm#L64, uses a single greater-than character ("`>`") in the `open()` call which indicates it will overwrite data.  If it used double greater-than characters ("`>>`") that would indicate it is appending.

Reply:
done